request-server_test.go

```diff
@@ -147,7 +147,7 @@ func TestRequestWriteEmpty(t *testing.T) {
 	f, err := r.fetch("/foo")
 	if assert.Nil(t, err) {
 		assert.False(t, f.isdir)
-		assert.Equal(t, f.content, []byte(""))
+		assert.Equal(t, []byte(""), f.content)
```

Comment:
Probably one should prefer `assert.Equal(t, len(f.content), 0)` to avoid making a possible distinction between a `nil` and an empty byte-slice.

Also, go prefers the order `got COMPARATOR want`  https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Some languages can easily confuse an assignment with a comparator, and so it became stylish to put your test value on the left because it was not a valid assignment L-value. But since go does not allow assignments in conditions, that risk is entirely removed, and the unconventional ordering just to make sure the L-value is unassignable is unnecessary.

Reply:
I seem to have an occasional bout of comparator dyslexia as I just am sure the reverse order is correct sometimes. I'll replace it with `assert.Len(t, f.content, 0)` that way I don't need to worry about the nil comparison.